tree/ntuple/v7/src/RPageStorageFile.cxx

```diff
@@ -0,0 +1,1414 @@
+      offsetof(struct RTFStreamerVersion, fNewClassTag) + 2;
+   streamerInfo.fStreamerInfo.fStreamers.fStreamerSize.fClassTag = 0x80000000 | classTagOffset;
+   streamerInfo.fStreamerInfo.fStreamers.fStreamerSeekHeader.fClassTag = 0x80000000 | classTagOffset;
+   streamerInfo.fStreamerInfo.fStreamers.fStreamerNBytesHeader.fClassTag = 0x80000000 | classTagOffset;
+   streamerInfo.fStreamerInfo.fStreamers.fStreamerLenHeader.fClassTag = 0x80000000 | classTagOffset;
+   streamerInfo.fStreamerInfo.fStreamers.fStreamerSeekFooter.fClassTag = 0x80000000 | classTagOffset;
+   streamerInfo.fStreamerInfo.fStreamers.fStreamerNBytesFooter.fClassTag = 0x80000000 | classTagOffset;
+   streamerInfo.fStreamerInfo.fStreamers.fStreamerLenFooter.fClassTag = 0x80000000 | classTagOffset;
+   streamerInfo.fStreamerInfo.fStreamers.fStreamerReserved.fClassTag = 0x80000000 | classTagOffset;
+   WriteKey(&streamerInfo, streamerInfo.GetSize(), fControlBlock->fHeader.GetSeekInfo(), 100, 1,
+            "TList", "StreamerInfo", "Doubly linked list");
+   fControlBlock->fHeader.SetNbytesInfo(fFilePos - fControlBlock->fHeader.GetSeekInfo());
+
+   // Reserve the space for the RNTuple record, which will be written on commit
+   fControlBlock->fSeekNTuple = fFilePos;
+   RTFKey keyRNTuple(fControlBlock->fSeekNTuple, 100, strRNTupleClass, strRNTupleName, strEmpty,
+                     fControlBlock->fNTuple.GetSize());
+
+   // The key index of the root TFile object, containing for the time being only the RNTuple key
+   fileRoot.fInfoShort.fSeekKeys = fFilePos + keyRNTuple.GetSize();
+   RTFKeyList keyList{1};
+   RTFKey keyKeyList(fileRoot.GetSeekKeys(), 100, strEmpty, strEmpty, strEmpty, keyList.GetSize() + keyRNTuple.fKeyLen);
+   Write(&keyKeyList, keyKeyList.fKeyHeaderSize, fileRoot.GetSeekKeys());
+   Write(&strEmpty, strEmpty.GetSize());
+   Write(&strEmpty, strEmpty.GetSize());
+   Write(&strEmpty, strEmpty.GetSize());
+   Write(&keyList, keyList.GetSize());
+   Write(&keyRNTuple, keyRNTuple.fKeyHeaderSize);
+   Write(&strRNTupleClass, strRNTupleClass.GetSize());
+   Write(&strRNTupleName, strRNTupleName.GetSize());
+   Write(&strEmpty, strEmpty.GetSize());
+   fileRoot.fNBytesKeys = fFilePos - fileRoot.GetSeekKeys();
+
+   Write(&keyRoot, keyRoot.fKeyHeaderSize, 100);
+   Write(&strTFile, strTFile.GetSize());
+   Write(&strFileName, strFileName.GetSize());
+   Write(&strEmpty, strEmpty.GetSize());
+   Write(&strFileName, strFileName.GetSize());
+   Write(&strEmpty, strEmpty.GetSize());
+   Write(&fileRoot, fileRoot.GetSize());
+
+   fControlBlock->fNTuple.fSeekHeader = fileRoot.GetSeekKeys() + fileRoot.fNBytesKeys;
+}
+
+
+void ROOT::Experimental::Detail::RPageSinkFile::DoCreate(const RNTupleModel & /* model */)
+{
+   fControlBlock = std::make_unique<ROOT::Experimental::Internal::RTFileControlBlock>();
+   const auto &descriptor = fDescriptorBuilder.GetDescriptor();
+   auto szHeader = descriptor.SerializeHeader(nullptr);
```

Comment:
This (doing serialization on nullptr to get the size) implies that all the low level routine all contains ```if (buffer != nullptr) {```.  Maybe (later) we can optimize this by making the Serialize* routine templated on whether they are counting or not (giving also the opportunity to do some of the counting at compile time).

Reply:
ping.

Reply:
I added a [TODO](https://github.com/root-project/root/pull/4674/commits/98d53d59e96aa110ef105ba7537069af2b069442)